config/v0.3.0/config_yaml.go

```diff
@@ -30,6 +31,27 @@ func (j *AttributeNameValueType) UnmarshalYAML(unmarshal func(interface{}) error
 	return nil
 }
 
+// UnmarshalYAML implements json.Unmarshaler.
+func (j *NameStringValuePair) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var raw map[string]interface{}
+	if err := unmarshal(&raw); err != nil {
+		return err
+	}
+	if _, ok := raw["name"]; raw != nil && !ok {
+		return errors.New("yaml: cannot unmarshal field name in NameStringValuePair required")
+	}
+	if _, ok := raw["value"]; raw != nil && !ok {
+		return errors.New("yaml: cannot unmarshal field value in NameStringValuePair required")
+	}
+	type Plain NameStringValuePair
+	var plain Plain
+	if err := unmarshal(&plain); err != nil {
+		return err
+	}
```

Comment:
Calling `unmarshal` seems quite inefficient. My not just use the already unmashaled `raw` to build `Plain`?

Reply:
it's a good question, this code is originally from the code generated with go-jsonschema, it could use some additional cleanups since we're not autogenerating the unmarshaling code anymore

Reply:
tried fixing this by using the raw fields and ensuring the raw types match what's expected. ptal